Code/GraphMol/FileParsers/testMultithreadedMolSupplier.cpp

```diff
@@ -113,9 +190,83 @@ void testSmiCorrectness() {
   expectedResult = 50000;
   testSmiConcurrent(path, " \t", 0, 1, false, true, 4, 1000, 100,
                     expectedResult);
+  /*
+          TEST PROPERTIES
+  */
+  testSmiProperties();
+}
+
+void testSDProperties() {
+  std::string rdbase = getenv("RDBASE");
+  std::string fname =
+      rdbase + "/Code/GraphMol/FileParsers/test_data/NCI_aids_few.sdf";
+  std::vector<std::string> nameVector;
+  std::string tempStr;
+  SDMolSupplier sup(fname, false);
+  MultithreadedSDMolSupplier multiSup(fname, false);
+
+  while (!multiSup.atEnd()) {
+    ROMol* mol = multiSup.next();
+    if (mol != nullptr) {
+      TEST_ASSERT(mol->hasProp(common_properties::_Name));
+      mol->getProp(common_properties::_Name, tempStr);
+      nameVector.push_back(tempStr);
+      TEST_ASSERT(mol->hasProp("NCI_AIDS_Antiviral_Screen_Conclusion"));
+      TEST_ASSERT(mol->hasProp("CAS_RN"));
+      TEST_ASSERT(mol->hasProp("NSC"));
+    }
+  }
+
+  while (!sup.atEnd()) {
+    ROMol* mol = sup.next();
+    if (mol != nullptr) {
+      mol->getProp(common_properties::_Name, tempStr);
+      TEST_ASSERT(std::find(nameVector.begin(), nameVector.end(), tempStr) !=
+                  nameVector.end());
+    }
+  }
+}
+
+void testSDCorrectness() {
+  /*
+          TEST CORRECTNESS
+  */
+
+  std::string path = "/Code/GraphMol/FileParsers/test_data/NCI_aids_few.sdf";
+  unsigned int expectedResult = 16;
+  testSDConcurrent(path, false, true, true, 2, 5, 5, expectedResult);
+
+  path = "/Code/GraphMol/FileParsers/test_data/outNCI_few_molsupplier.sdf";
+  expectedResult = 16;
+  testSDConcurrent(path, true, true, true, 2, 5, 5, expectedResult);
+
+  path = "/Code/GraphMol/FileParsers/test_data/esters_end.sdf";
+  expectedResult = 6;
+  testSDConcurrent(path, true, true, true, 2, 5, 5, expectedResult);
+
+  // strict parsing results in reading 0 out of 2 records
+  path = "/Code/GraphMol/FileParsers/test_data/strictLax1.sdf";
+  expectedResult = 0;
+  testSDConcurrent(path, true, true, true, 2, 5, 5, expectedResult);
+
+  expectedResult = 2;
+  testSDConcurrent(path, true, true, false, 2, 5, 5, expectedResult);
+
+  path = "/Regress/Data/O2.sdf";
+  expectedResult = 1;
+  testSDConcurrent(path, true, true, false, 2, 5, 5, expectedResult);
+
+  path = "/Regress/Data/mols1000.sdf";
```

Comment:
We should talk about what to do with this somewhat larger input file. It's definitely good to do a test with 1000 molecules, but adding a 6.5MB SDF to git isn't the nicest thing.

Reply:
the same comment applies to znp.50k.smi